src/test/regress/expected/master_evaluation.out

```diff
@@ -119,11 +132,402 @@ SELECT min(get_local_node_id_volatile())  > 0 FROM master_evaluation_table;
+DEBUG:  Router planner cannot handle multi-shard select queries
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+EXECUTE router_select_with_param_expression(0);
+DEBUG:  Router planner cannot handle multi-shard select queries
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+EXECUTE router_select_with_param_expression(0);
+DEBUG:  Router planner cannot handle multi-shard select queries
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+PREPARE router_select_with_param(int) AS SELECT DISTINCT value FROM master_evaluation_table WHERE key = $1;
+-- this time the parameter itself is a function, so should be evaluated
+-- on the coordinator
+EXECUTE router_select_with_param(0 + get_local_node_id_volatile());
+DEBUG:  Deferred pruning for a fast-path router query
+DEBUG:  Creating router plan
+DEBUG:  Plan is router executable
+ value
+---------------------------------------------------------------------
+     0
+(1 row)
+
+EXECUTE router_select_with_param(0 + get_local_node_id_volatile());
+DEBUG:  Deferred pruning for a fast-path router query
+DEBUG:  Creating router plan
+DEBUG:  Plan is router executable
+ value
+---------------------------------------------------------------------
+     0
+(1 row)
+
+EXECUTE router_select_with_param(0 + get_local_node_id_volatile());
+DEBUG:  Deferred pruning for a fast-path router query
+DEBUG:  Creating router plan
+DEBUG:  Plan is router executable
+ value
+---------------------------------------------------------------------
+     0
+(1 row)
+
+EXECUTE router_select_with_param(0 + get_local_node_id_volatile());
+DEBUG:  Deferred pruning for a fast-path router query
+DEBUG:  Creating router plan
+DEBUG:  Plan is router executable
+ value
+---------------------------------------------------------------------
+     0
+(1 row)
+
+EXECUTE router_select_with_param(0 + get_local_node_id_volatile());
+DEBUG:  Deferred pruning for a fast-path router query
+DEBUG:  Creating router plan
+DEBUG:  Plan is router executable
+ value
+---------------------------------------------------------------------
+     0
+(1 row)
+
+EXECUTE router_select_with_param(0 + get_local_node_id_volatile());
+DEBUG:  Deferred pruning for a fast-path router query
+DEBUG:  Creating router plan
+DEBUG:  Plan is router executable
+ value
+---------------------------------------------------------------------
+     0
+(1 row)
+
+EXECUTE router_select_with_param(0 + get_local_node_id_volatile());
+ value
+---------------------------------------------------------------------
+     0
+(1 row)
+
+-- same calls with functions as the parametres only
+EXECUTE router_select_with_param(get_local_node_id_volatile());
+ value
+---------------------------------------------------------------------
+     0
+(1 row)
+
+EXECUTE router_select_with_param(get_local_node_id_volatile());
```

Comment:
This is  actually  one class of the queries  that I'm not sure what is the correct  behaviour.
This is the same as 
```SQL
prepare p2(int) AS SELECT count(*) FROM master_evaluation_table WHERE key = $1;
 execute p2(random());
```

In this patch,  and  in 9.1 as well , Citus evaluates `random` as  it  is a parameter.  I think it is probably not the correct behaviour. Given that it is also similar in  9.1 (and probably way back), I'm planning to punt on this for now.

Any comment?

Reply:
Is it actually Citus that evaluates random() or postgres? Seems like the correct behaviour in any case.

Reply:
> Is it actually Citus that evaluates random() or postgres?

Citus does

> Seems like the correct behaviour in any case.

Ok, verified that Postgres also behaves the same.